beacon-chain/das/iface.go

```diff
@@ -0,0 +1,19 @@
+package das
+
+import (
+	"context"
+
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+)
+
+// AvailabilityStore describes a component that can verify and save sidecars for a given block, and confirm previously
+// verified and saved sidecars.
+// Persist guarantees that the sidecar will be available to perform a DA check
+// for the life of the beacon node process.
+// IsDataAvailable guarantees that all blobs committed to in the block have been
+// durably persisted before returning a non-error value.
+type AvailabilityStore interface {
+	IsDataAvailable(ctx context.Context, current primitives.Slot, b blocks.ROBlock) error
+	Persist(current primitives.Slot, sc ...blocks.ROBlob) error
```

Comment:
What do you think about passing in the current slot versus assuming the implementation that satisfies these interfaces can get its current slot? Like`s.CurrentSlot()`. I'm not suggesting a change but rather curious

Reply:
This was a decision in the older branch so I'm trying to remember my reasoning - I'm pretty sure it was because I wanted to avoid infecting this code with the whole `WaitForClock` mess that would come with giving it a clock. When we get that initialization path refactored to remove the need for clock waiting I think we could easily change this to keep an internal `CurrentSlotter`.